app/Domain/Users/Controllers/EditOwn.php

```diff
@@ -254,5 +262,37 @@ public function post(): void
             //Redirect
             FrontcontrollerCore::redirect(BASE_URL . "/users/editOwn" . $tab);
         }
+
+        /**
+         * Returns list of supported varying date-time formats.
+         * @link https://www.php.net/manual/en/class.datetimeinterface.php#datetimeinterface.constants.types
+         * 
+         * @return [<string,string>] Format of ID => date-time string
+         */
+        private function getSupportedDateTimeFormats(): array
+        {
+            return [
+                'dates' => [
+                    $this->language->__("language.dateformat"),
+                    'Y-m-d',
```

Comment:
Is this working for you? When I try to save an updated string I get a fatal error that the array index cannot be found on editOwn.

Reply:
Are you sure you have the latest changes?  I was able to switch formats just fine and render them both with PHP calls and JS.

Reply:
Let me double check

Reply:
Now I'm getting a fatal error:

[28-Oct-2023 11:25:09 America/Los_Angeles] 2023-10-28T11:25:09-07:00 [critical] Uncaught Target class [Leantime\Core\ServicesSetting] does not exist. (View: /Volumes/Source/Sites/leantime-oss/app/Views/Templates/layouts/entry.blade.php)
[28-Oct-2023 11:25:11 America/Los_Angeles] 2023-10-28T11:25:11-07:00 [critical] Uncaught Target class [Leantime\Core\ServicesSetting] does not exist. (View: /Volumes/Source/Sites/leantime-oss/app/Views/Templates/layouts/entry.blade.php)

Reply:
Seems to be link 602 in language: 

        $settings = app()->make(ServicesSetting::class);

Reply:
After instantiating the right class and trying to navigate to editOwn, I get this. Can you pull from master and check if everything is working on your end? 

<img width="1087" alt="image" src="https://github.com/Leantime/leantime/assets/47038998/68cd9839-ecbf-4bc2-979c-9e3e7427f019">

Reply:
Sure I'll check that tomorrow.  Makes sense now that I think about it as to how you were having issues.

Reply:
Thank you

Reply:
Pushed up 44a77f8 as somehow the template setters got removed.  I tested with the settings deleted from the DB (so a user who hasn't set this before) and they rendered properly, and I was able to save them.

Reply:
Thanks. I made a couple fixes and created PR into your branch. 
1. It seems you were pulling directly from the db every time a dateformat needed to be output. I went ahead and cached the result in the session
2. Timeformat was not returned
3. Added missing language strings to en-US.

Please take a look and merge into your branch (which should update this PR here automatically.